detect_secrets/filters/heuristic.py

```diff
@@ -165,19 +165,19 @@ def is_indirect_reference(secret: str) -> bool:
 
         secret = request.headers['apikey']
     """
-    output = False
-    for start, end in (
-        list('()'),
-        list('[]'),
-    ):
-        try:
-            output = secret.index(start) < secret.index(end)
-            if output:
-                return output
-        except ValueError:
-            continue
-
-    return output
+    # Regex details:
+    #   ([^\v=!:]*)     ->  Something before the assignment or comparison
+    #   \s*             ->  Some optional whitespaces
+    #   (:=?|[!=]{1,3}) ->  Assignment or comparison: :=, =, ==, ===, !=, !==
+    #   \s*             ->  Some optional whitespaces
+    #   (
+    #       [\w.-]+     ->  Some alphanumeric character, dot or -
+    #       [\[\(]      ->  Start of indirect reference: [ or (
+    #       [^\v]*      ->  Something except line breaks
+    #       [\]\)]      ->  End of indirect reference: ] or )
+    #   )
+    regex = re.compile(r'([^\v=!:]*)\s*(:=?|[!=]{1,3})\s*([\w.-]+[\[\(][^\v]*[\]\)])')
```

Comment:
Let's cache this regex, as described in [Best Practices for Writing Filters](https://github.com/Yelp/detect-secrets/blob/master/docs/filters.md#best-practices). An example is found in that documentation too (search `Example #2`)

Reply:
Okay, perfect